src/test/java/seedu/mark/logic/parser/ParserUtilTest.java

```diff
@@ -128,6 +131,40 @@ public void parseUrl_validValueWithWhitespace_returnsTrimmedUrl() throws Excepti
         assertEquals(expectedUrl, ParserUtil.parseUrl(urlWithWhitespace));
     }
 
+    @Test
+    public void parseTab_validIndex_success() throws Exception {
+        TabCommand.Tab expected = TabCommand.Tab.ONLINE;
+        assertEquals(expected, ParserUtil.parseTabIndex("2"));
+        expected = TabCommand.Tab.OFFLINE;
+        assertEquals(expected, ParserUtil.parseTabIndex("3"));
+        expected = TabCommand.Tab.DASHBOARD;
+        assertEquals(expected, ParserUtil.parseTabIndex("1"));
+    }
+
+    @Test
+    public void parseTab_invalidIndex_throwsInvalidIndexParseException() {
+        try {
+            ParserUtil.parseTabIndex("5");
+            ParserUtil.parseTabIndex("0.5");
+            fail();
+        } catch (ParseException pe) {
+            assertEquals(new ParseException(TabCommand.MESSAGE_INVALID_INDEX).getMessage(), pe.getMessage());
+        }
+    }
+
+    @Test
+    public void parseTab_invalidArg_throwsInvalidFormatParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseTabIndex("invalid arg"));
+        try {
+            ParserUtil.parseTabIndex("invalid arg");
+            fail();
```

Comment:
uh what are these fail things for? :o

Reply:
I put the fail there so if somehow no exception is thrown then the test fails

Reply:
not supposed to put it there isit?

Reply:
Ya, and I think you should change it to assertThrows ( some examples in the code)

Reply:
Okok I go change